setup.py

```diff
@@ -9,143 +9,44 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from __future__ import annotations
-
-import glob
 import os
-import re
-import sys
 import warnings
 
-import pkg_resources
 from setuptools import find_packages, setup
 
 import versioneer
 
-# TODO: debug mode -g -O0, compile test cases
-
-RUN_BUILD = os.getenv("BUILD_MONAI", "0") == "1"
-FORCE_CUDA = os.getenv("FORCE_CUDA", "0") == "1"  # flag ignored if BUILD_MONAI is False
-
-BUILD_CPP = BUILD_CUDA = False
-TORCH_VERSION = 0
-try:
-    import torch
-
-    print(f"setup.py with torch {torch.__version__}")
-    from torch.utils.cpp_extension import BuildExtension, CppExtension
-
-    BUILD_CPP = True
-    from torch.utils.cpp_extension import CUDA_HOME, CUDAExtension
-
-    BUILD_CUDA = FORCE_CUDA or (torch.cuda.is_available() and (CUDA_HOME is not None))
+# Determine whether to build C++ and CUDA extensions
+BUILD_MONAI = os.getenv("BUILD_MONAI", "0") == "1"
+FORCE_CUDA = os.getenv("FORCE_CUDA", "0") == "1"
 
-    _pt_version = pkg_resources.parse_version(torch.__version__).release
-    if _pt_version is None or len(_pt_version) < 3:
-        raise AssertionError("unknown torch version")
-    TORCH_VERSION = int(_pt_version[0]) * 10000 + int(_pt_version[1]) * 100 + int(_pt_version[2])
-except (ImportError, TypeError, AssertionError, AttributeError) as e:
-    warnings.warn(f"extension build skipped: {e}")
-finally:
-    if not RUN_BUILD:
-        BUILD_CPP = BUILD_CUDA = False
-        print("Please set environment variable `BUILD_MONAI=1` to enable Cpp/CUDA extension build.")
-    print(f"BUILD_MONAI_CPP={BUILD_CPP}, BUILD_MONAI_CUDA={BUILD_CUDA}, TORCH_VERSION={TORCH_VERSION}.")
-
-
-def torch_parallel_backend():
```

Comment:
This is needed for choosing OMP compiler flags and macros farther down, I don't see this being done in your version and I think we do need them. Further down in the original `get_extensions()` uses this and `omp_flags()` to set compiler flags and the matrco for the backend type.